### PR TITLE
[Mentions v2] Add new rule in ExpensiMark for room mentions

### PR DESCRIPTION
**Pull request by robertKozik**
> [!IMPORTANT]
> Room mentions consists of `#` symbol followed by room name. Room name consist of any char sequence connected by hyphens (no spaces is allowed inside room name)

### Fixed Issues
$ https://github.com/Expensify/App/issues/39538

# Tests
1. What unit/integration tests cover your change? What autoQA tests cover your change?
I've created multiple unit tests to cover most of the possible usages of room/report mentions. Room mentions consists of `#` symbol followed by room name 
1. What tests did you perform that validates your changed worked?
Check if existing unit tests are not failing after my. changes, added additional tests to cover room mentions

# QA
1. What does QA need to do to validate your changes?
Room mentions are not yet visible inside E/App, so to validate this changes we should only check for regressions or other markdown rules
1. What areas to they need to test for regressions?
Room mention syntax is very similar to Heading syntax. Extensive testing of this aspeect of markdown parser is recommended. Moreover all other markdown rules should be untouched

### CODE REVIEW

File: lib/ExpensiMark.js [hunk 1]
```diff
@@ -169,6 +169,13 @@ export default class ExpensiMark {
                 },
             },
 
+            {
```
Comment (rlinoz) [inline lib/ExpensiMark.js:172]: Can we add a comment here explaining the regex the same way the other mention regex have?

File: __tests__/ExpensiMark-HTML-test.js [hunk 2]
```diff
@@ -1959,5 +1959,92 @@ describe('Image markdown conversion to html tag', () => {
         const testString = '![`code`](https://example.com/image.png)';
         const resultString = '<img src="https://example.com/image.png" alt="<code>code</code>" />';
         expect(parser.replace(testString)).toBe(resultString);
-    })
+    });
+});
+
+describe('room mentions', () => {
```
Comment (rlinoz) [inline __tests__/ExpensiMark-HTML-test.js:1965]: I think we can add a test for:
- `#` without following characters, should yield no mention
- Room name with numbers
- Room name with hyphen

**Review by rlinoz — CHANGES_REQUESTED** [review]

**Comment by robertKozik** [comment]
Thanks @rlinoz, I've added tests and comment

File: __tests__/ExpensiMark-HTML-test.js [hunk 3]
```diff
@@ -1959,5 +1959,110 @@ describe('Image markdown conversion to html tag', () => {
...
+        expect(parser.replace(testString)).toBe(resultString);
+    });
+
+    test('room mention with space between hash and room name', () => {
+        const testString = '# room';
+        const resultString = '<h1>room</h1>';
+        expect(parser.replace(testString)).toBe(resultString);
+    });
+
+    test.only('room mention with markdown syntax before the # prefix', () => {
```
Comment (Beamanator) [inline __tests__/ExpensiMark-HTML-test.js:1990]: Did you mention to have `.only` here? I'm not entirely sure what this is used for, but it looks like we don't do that anywhere else in our tests in this repo 🤔 

```suggestion
    test('room mention with markdown syntax before the # prefix', () => {
```

**Review by Beamanator — CHANGES_REQUESTED** [review]
Can we also add at least 1 test for room names with `-` since that will probably be a pretty common use-case? Ex: `#room-name`, maybe even `#room-123`

File: __tests__/ExpensiMark-HTML-test.js (hunk 3 above)
Comment (robertKozik) [reply __tests__/ExpensiMark-HTML-test.js]: Yeah, I've commited it by mistake. Thanks for noticing it 🙇🏼

Comment (Beamanator) [reply __tests__/ExpensiMark-HTML-test.js]: Cool, just checking! Thanks for the quick fix!

**Review by Beamanator — APPROVED** [review]
Looking good to me!

**Review by rlinoz — APPROVED** [review]

File: lib/ExpensiMark.js [hunk 4]
```diff
@@ -169,6 +169,19 @@ export default class ExpensiMark {
...
+            /**
+             * A room mention is a string that starts with the '#' symbol and is followed by a valid room name.
+             *
+             * Note: We are allowing mentions in a format of #room-name The room name can contain any
+             * combination of letters and hyphens
+             */
+            {
+                name: 'roomMentions',
+
+                regex: /(?<![^ \n*~_])(#[\p{Ll}0-9-]{1,80})/gmiu,
```
Comment (rojiphil) [inline lib/ExpensiMark.js:181]: NAB but certain improvements were introduced as mentioned in [this issue](https://github.com/Expensify/App/issues/46299) with respect to identifying room mentions.

**Pull request merged**